src/pages/MangaHub/main.ts

```diff
@@ -0,0 +1,142 @@
+import { pageInterface } from '../pageInterface';
+
+export const MangaHub: pageInterface = {
+  name: 'MangaHub',
+  domain: 'https://mangahub.io',
+  languages: ['English'],
+  type: 'manga',
+  isSyncPage(url) {
+    if (j.$('#mangareader').length) {
+      return true;
+    }
+    return false;
+  },
+  sync: {
+    getTitle(url) {
+      return j
+        .$('div._3_X6m.container-fluid h3 a')
+        .text()
+        .trim();
+    },
+    getIdentifier(url) {
+      return utils.urlPart(url, 4);
+    },
+    getOverviewUrl(url) {
+      return j.$('div._3_X6m.container-fluid h3 a').attr('href') || '';
+    },
+    getEpisode(url) {
+      const episodePart = utils.urlPart(url, 5);
+
+      const temp = episodePart.match(/chapter-\d+/gim);
+
+      if (!temp || !temp.length) return 0;
+
+      return Number(temp[0].replace(/\D+/g, ''));
+    },
+    nextEpUrl(url) {
+      return $('ul.dropdown-menu li.active')
+        .next()
+        .find('a')
+        .attr('href');
+    },
+  },
+  overview: {
+    getTitle(url) {
+      return j
+        .$('div._3QCtP.col-md-9.col-sm-8.col-xs-6 h1')
+        .clone()
+        .children()
+        .remove()
+        .end()
+        .text();
+    },
+    getIdentifier(url) {
+      return utils.urlPart(url, 4);
+    },
+    uiSelector(selector) {
+      j.$('section._2fecr').after(
+        j.html(
+          `<section class="_2fecr"><div style="background-color: inherit;" class="container-fluid"><div class="row" style="background-color: inherit;"><div class="col-md-1"><span style="font-weight: 700;">MALSync:</span></div><div class="col-md-11" style="background-color: inherit;">${selector}</div></div></div></section>`,
+        ),
+      );
+    },
+    list: {
+      offsetHandler: false,
+      elementsSelector() {
+        return j.$('#noanim-content-tab div').find('li');
+      },
+      elementUrl(selector) {
+        return (
+          selector
+            .find('a')
+            .first()
+            .attr('href') || ''
+        );
+      },
+      elementEp(selector) {
+        return MangaHub.sync.getEpisode(MangaHub.overview!.list!.elementUrl(selector));
+      },
+    },
+  },
+  init(page) {
+    api.storage.addStyle(require('!to-string-loader!css-loader!less-loader!./style.less').toString());
+    j.$(document).ready(function() {
+      utils.urlChangeDetect(() => {
+        utils.waitUntilTrue(
+          function() {
+            return !j.$('html.no-js.nprogress-busy').length;
+          },
+          function() {
+            page.reset();
+            if (j.$('#mangareader').length) {
+              utils.waitUntilTrue(
```

Comment:
This could result in triggering multiple times at the same time. The function returns a interval that could be destroyed before setting an new wait.
https://github.com/MALSync/MALSync/blob/e8125839ee0e92a3f7af806342f2da626c67663d/src/pages/DreamAnimes/main.ts#L58-L59

This and the comment before could be fixed for the popup of multiple UI on the overview of an anime.
Ex.:
open https://mangahub.io/manga/no-game-no-life
Click on one "Popular Manga Updates" on the bottom of the page. Now the ui should show multiple times.
![image](https://user-images.githubusercontent.com/12599334/109983973-a3b73980-7d03-11eb-95ce-06e58bb11b1e.png)